lib/HLSL/DxilValidation.cpp

```diff
@@ -4221,8 +4214,10 @@ static void ValidateShaderFlags(ValidationContext &ValCtx) {
     return;
   }
   ValCtx.EmitError(ValidationRule::MetaFlagsUsage);
-  ValCtx.DiagStream() << "Flags declared=" << declaredFlagsRaw
-                      << ", actual=" << calcFlagsRaw << "\n";
+
+  std::string Note = "Flags declared=" + declaredFlagsRaw;
+  Note += ", actual=" + calcFlagsRaw;
+  dxilutil::EmitNoteOnContext(ValCtx.M.getContext(), Note);
```

Comment:
First, I don't think this code does what you think it does!  Adding the uint64_t value to literal string is going to offset the pointer, not convert the number to a string and concatenate that.
Since EmitNoteOnContext takes Twine, why not just use that:
```
  dxilutil::EmitNoteOnContext(ValCtx.M.getContext(),
    Twine("Flags declared=") + Twine(declaredFlagsRaw) +
    Twine(", actual=") + Twine(calcFlagsRaw));
```